app/assets/javascripts/discourse/widgets/topic-map.js.es6

```diff
@@ -148,7 +148,16 @@ createWidget('topic-map-link', {
       content = `${content.substr(0, truncateLength).trim()}...`;
     }
 
-    return attrs.title ? replaceEmoji(content) : content;
+    let categoriesRegexp = /\/c\/[0-9]*-(.*)/;
+    content = attrs.title ? replaceEmoji(content) : content;
+
+    let matches = categoriesRegexp.exec(content);
+    if(matches) {
+      return content.replace(categoriesRegexp, '$1 Category').replace(/-/g, ' ').replace(/(^| )(\w)/g, function (x) {
```

Comment:
This will only work for English. Better solution would be to try to retrieve the name of the category from the list of categories.

Reply:
Oh yes, I'll try to do it this way.